chore: add api for read unread (#1383)
WalletConnect/walletconnect-docs: docs/web3inbox/frontend-integration/api.mdx

// watch notifications of current account's subscription to current dapp
const notificationsPerPage = 5
const isInfiniteScroll = true

> const unreadFirst = true

Reviewer: still not sure how unread first should work

Author: the way I understood it, notify would order them separately and place them in the front.

instead of  hypothetically instead of just sorting notification array by time sent, it would first sort unread notifications, then sort read notifications and then concact those two into one array `[...unreadNotifics, ...readNotifis]`

So if hypothetically, you had an unread notif from december 2023, and all your notifications except that one are read, including a notif you got today, the unread one would still show up first

const onUpdate = ({notifications: fetchedNotifications}: GetNotificationsReturn) => {
  notifications = fetchedNotifications
}

const { nextPage, markAllReturnedNotificationsAsRead, markNotificationAsRead } = client.pageNotifications(
  notificationsPerPage,
  isInfiniteScroll,
  unreadFirst,
)(onUpdate)

// marking a single notification as read
notifications[0].markAsRead();

// mark all notifications currently in `notifications` returned array as read
markAllReturnedNotificationsAsRead();

// mark specific notifications as read
markNotificationsAsRead(notifications.slice(2).map(n => n.id))
```

#### References

- **notificationsPerPage:** Number representing how many notifications to get per fetch
- **isInfiniteScroll:** Whether or not to keep already fetched notifications when getting next page
- **unreadFirst:** Whether or not unread messages should be sorted at the top, regardless of timestamp
- **onUpdate:**: A callback that will be called whenever notifications get updated
- **nextPage:**: A function to be called to fetch the next page of notifications
- **notifications:** Array of notifications, of type
- **notification.markAsRead:** Mark the notification as read

> - **markAllReturnedNotificationsAsRead**: Mark currently fetched notifications as read. *This does not mark all notifications on notify server as read, only those available on the client*.

Reviewer 2: Shouldn't get too technical about the "notify server"

```suggestion
- **markAllReturnedNotificationsAsRead**: Mark currently fetched notifications as read. *This does not mark all notifications as read, just the ones that have been fetched and returned here.*.
```
